example/build_cmake_in_tree/CMakeLists.txt

```diff
@@ -1,44 +1,24 @@
-# Kokkos requires CMake version 3.1 or higher and that you have the following
-# line with a version of 3.1 or higher as the first line of your project:
-#   cmake_minimum_required(VERSION 3.1)
-#
-# The other CMake commands required to build Kokkos as part of your application
-# are:
-#   add_subdirectory(path/to/kokkos)
-#   target_link_libraries(executable or library)
-#
-# If Kokkos is not a subdirectory of your project, you will also need to pass a
-# binary directory to add_subdirectory().  We had to pass the binary directory
-# for this example for that reason.  Note that target_link_libraries() can be
-# called on a target added by add_executable(), add_library(), or another
-# similar command.
-#
-# All the flags, etc. required to build using the Kokkos library are
-# transitively added to targets which depend on the library.
-#
-# The CMake variables CMAKE_CXX_STANDARD and CMAKE_CXX_EXTENSIONS are
-# respected.  We recommend that you set CMAKE_CXX_EXTENSIONS to OFF.
-# Otherwise, CMake defaults to using extensions for the C++ standard, and the
-# GNU extensions (-std=gnu++11) will be used for compilers that support it
-# instead of standard C++11 (-std=c++11).
-#
-# A bunch of build options are added as variables (all starting with KOKKOS_)
-# to the build.  Check them out using ccmake or the CMake GUI.
-#
-# Building this example:
-#   1. Create a build directory.
-#   2. cd /path/to/build/directory
-#   3. cmake /path/to/example
-#   4. make
+# Kokkos minimally requires 3.10 right now,
+# but your project can set it higher
+cmake_minimum_required(VERSION 3.10)
 
-cmake_minimum_required(VERSION 3.1)
+# Declare your project - must have C++ support
 project(Example CXX)
 
-list(APPEND CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS} -O3)
-
-add_subdirectory(${Example_SOURCE_DIR}/../.. ${Example_BINARY_DIR}/kokkos)
-
-include_directories(${Kokkos_INCLUDE_DIRS_RET})
+if (Kokkos_INSTALL_TESTING)
+  # Look for an installed Kokkos
+  if(CMAKE_VERSION VERSION_GREATER_EQUAL "3.12.0")
+    message(STATUS "Setting policy CMP0074 to use <Package>_ROOT variables")
+    cmake_policy(SET CMP0074 NEW)
+  endif()
+  find_package(Kokkos REQUIRED)
+else()
+  # In this case, we build Kokkos in-tree
+  add_subdirectory(${Example_SOURCE_DIR}/../.. ${Example_BINARY_DIR}/kokkos)
+endif()
 
 add_executable(example cmake_example.cpp)
-target_link_libraries(example kokkos)
+target_link_libraries(example Kokkos::kokkos)
+
+enable_testing()
+add_test(NAME KokkosInTree_Verify COMMAND example 10)
```

Comment:
I don't feel strongly about showing how to use `CTest` or not but if we do, it would probably make sense to also call `ctest` in `Jenkins`.